pkg/catalog/routes.go

```diff
@@ -514,9 +529,150 @@ func (mc *MeshCatalog) listMeshServices() []service.MeshService {
+	outPolicies := []*trafficpolicy.OutboundTrafficPolicy{}
+
+	// fetch services running workloads with destination service account
+	destServices, err := mc.destinationServicesFromTrafficTarget(t)
+	if err != nil {
+		log.Error().Msgf("Error resolving destination from traffic target %s (%s): %v", t.Name, t.Namespace, err)
+		return outPolicies
+	}
+
+	// build an outbound traffic policy for each destination service
+	for _, destService := range destServices {
+		hostnames, err := mc.getServiceHostnames(destService, source.Namespace == destService.Namespace)
+		if err != nil {
+			log.Error().Err(err).Msgf("Error getting service hostnames for service %s", destService)
+			continue
+		}
+		weightedCluster := getDefaultWeightedClusterForService(destService)
+
+		policy := trafficpolicy.NewOutboundTrafficPolicy(policyName(destService, source.Namespace == destService.Namespace), hostnames)
+		if err := policy.AddRoute(allowAllRouteMatch, weightedCluster); err != nil {
+			log.Error().Err(err).Msgf("Error adding Route to outbound policy for source %s(%s) and destination %s (%s)", source.Name, source.Namespace, destService.Name, destService.Namespace)
+			continue
+		}
+
+		outPolicies = append(outPolicies, policy)
+	}
+	return outPolicies
+}
+
+// listPoliciesFromTrafficTargets loops through all SMI Traffic Target resources and returns inbound and outbound traffic policies
+//		based on when the given service account matches a destination or source in the Traffic Target resource
+func (mc *MeshCatalog) listPoliciesFromTrafficTargets(sa service.K8sServiceAccount) ([]*trafficpolicy.InboundTrafficPolicy, []*trafficpolicy.OutboundTrafficPolicy, error) {
+	inboundPolicies := []*trafficpolicy.InboundTrafficPolicy{}
+	outboundPolicies := []*trafficpolicy.OutboundTrafficPolicy{}
+
+	for _, t := range mc.meshSpec.ListTrafficTargets() { // loop through all traffic targets
+		if !isValidTrafficTarget(t) {
+			continue
+		}
+
+		if t.Spec.Destination.Name == sa.Name { // found inbound
+			inboundPolicies = trafficpolicy.MergeInboundPolicies(inboundPolicies, mc.buildInboundPolicies(t)...)
+		}
+
+		for _, source := range t.Spec.Sources {
+			if source.Name == sa.Name && source.Namespace == sa.Namespace { // found outbound
+				mergedPolicies, mergeErrors := trafficpolicy.MergeOutboundPolicies(outboundPolicies, mc.buildOutboundPolicies(sa, t)...)
+				outboundPolicies = mergedPolicies
+				for _, mergeError := range mergeErrors {
+					log.Error().Err(mergeError).Msgf("Error building outbound policies for source %s (%s) and with traffic target %s (%s)", source.Name, source.Namespace, t.Name, t.Namespace)
+				}
+				break
+			}
+		}
+	}
+	return inboundPolicies, outboundPolicies, nil
+}
+
 func isValidTrafficTarget(t *target.TrafficTarget) bool {
 	if t.Spec.Rules == nil || len(t.Spec.Rules) == 0 {
 		return false
 	}
 	return true
 }
+
+func policyName(svc service.MeshService, sameNamespace bool) string {
```

Comment:
please add a verb to the function name: `get, set, etc.` unless they are methods on types (ex. `svc.String()`)